### PR TITLE
Make PBEM/PBF password storage optional (#1884)

### PR DESCRIPTION
**Pull request by ssoloff**
This PR partially addresses #1884 by giving the user the option to request their PBEM/PBF password not be saved locally.  If they do choose to save their password locally, a warning will be displayed informing them that the password will not be encrypted.

This is a short-term fix to address the vulnerability of storing a user's password unencrypted (by not storing it at all).  A follow-up PR will be submitted to encrypt the user's password if they choose to save it.

I did not modify the "Play by Website" code, as there is an open question as to whether or not this feature will be removed (#1885).

#### Functional changes
* All PBEM/PBF providers now give the user the option to save or not save their password locally via a new "Remember my password" checkbox that appears just below the Password field.  The following screenshot shows the new UI for the forums.triplea-game.org PBF and Gmail PBEM providers (the other providers are similar):
&nbsp;
pbem-ui
* The default value of "Remember my password" is not selected (`false`).  The user is now required to explicitly opt-in to use this feature (all other provider properties will be saved as before).  When the user selects the checkbox, a message is displayed warning them that their password will be stored unencrypted:
&nbsp;
pbem-warning
&nbsp;
* All PBEM/PBF help files were updated to reflect that password storage is now optional and that, if saved, the password will be unencrypted.

#### Functional issues for review
* Please review all UI text and suggest any better wording.  I think the security warning dialog text can be better, but I'm not sure what else to say here.
* I implemented the option to store the password by piggybacking on the existing feature that ensures passwords are not stored in save game files.  I'm not sure if this was the right thing to do, but it was straightforward and required the fewest changes.  I originally implemented the feature by hooking the Serializable `writeObject()` callback, but I got a bad smell after I had to backup and restore the password around the call to `ObjectOutputStream#defaultWriteObject()`.

#### Refactoring changes
None.

#### Testing
I started games using all six providers (three PBF and three PBEM) to ensure the user's password was or was not stored based on the state of the checkbox.  (I wasn't aware of this before, but the provider data is not written until you actually start a game.)  I also verified that restarting the game with the provider data in both of the possible states (password saved or not) correctly populated the UI fields (i.e. password field empty or not empty, and checkbox selected or not selected).

The only provider I tested end-to-end (meaning, within the game, I actually posted a saved game to the service) is the Gmail provider.  **If someone could test any of the PBF providers, that would be appreciated.**

I tested backwards compatibility by starting this branch with a `local.cache` file saved by a previous version (Gmail provider only).  The UI initialized without any error.  The password field contained the password loaded from `local.cache` and the checkbox was unselected.  Without modifying any of the data, I started a new PBEM game and verified that the `local.cache` file no longer contained my Gmail password.  Obviously, upon restarting TripleA, the Gmail password field was now empty.

### CODE REVIEW

File: src/main/resources/data/games/strategy/triplea/help/tripleaForum.html [hunk 1]
```diff
@@ -1,15 +0,0 @@
-<html>
```
Comment (ssoloff) [inline src/main/resources/data/games/strategy/triplea/help/tripleaForum.html]: For some reason, this file wasn't covered by #1564 (the corresponding PR may have been in-flight when #1564 was opened).  Anyway, that's why it was moved out of the `data` folder.

File: src/main/java/games/strategy/engine/framework/startup/ui/editors/EmailSenderEditor.java [hunk 2]
```diff
@@ -128,10 +136,23 @@ private void setupListeners() {
...
   }
 
+  private void passwordSavedChanged() {
+    fireEditorChanged();
+
+    if (passwordSaved.isSelected()) {
+      GameRunner.showMessageDialog(
+          "Your password will be stored unencrypted in the local file system. "
+              + "You should not choose to remember your password if this makes you uncomfortable.",
+          GameRunner.Title.of("Security Warning"),
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/framework/startup/ui/editors/EmailSenderEditor.java]: Really scary text to make the user nervous. :smile:  Please suggest improvements.  (Identical text appears in `ForumPosterEditor`.)

File: src/main/java/games/strategy/engine/pbem/GenericEmailSender.java [hunk 3]
```diff
@@ -173,10 +174,21 @@ public String getPassword() {
...
 
+  @Override
+  public boolean isPasswordSaved() {
+    return passwordSaved;
+  }
+
+  @Override
+  public void setPasswordSaved(final boolean passwordSaved) {
+    this.passwordSaved = passwordSaved;
+    setPassword(m_transPassword);
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/pbem/GenericEmailSender.java]: The purpose of this line is to ensure `m_password` will be updated appropriately if the `passwordSaved` field has changed.  (Identical code appears in `AbstractForumPoster`.)

**Comment by ron-murhammer** [comment]
Looks good especially as I consider this a temporary measure that might not ever get released anyways. I would very much like to have the fix in before we release the next version. Merging.

**Comment by ssoloff** [comment]
@ron-murhammer Hopefully, I'll have the permanent fix done next week.  Although, I do think we should keep the checkboxes to allow users to opt-out of persisting their passwords.  Even with encryption, we'll probably go with @RoiEXLab's suggestion to just store the key unencrypted (but at least in a secure location with permissions such that only the logged-in user can access it).  That still leaves an attack vector if the user's local account is ever compromised, and some users may still prefer to avoid any possibility of their email/forum passwords from being disclosed.

I also agree with @RoiEXLab that use of a master password to encrypt the key is probably overkill for TripleA.  If the user is overly conscious about security, they'll opt-out of TripleA storing the password and use their own password manager application (presumably with its own master password/token/etc. protection).

**Pull request merged**